### DISCUSSION TITLE
How to prefetch dependent queries

### DISCUSSION THREAD

**Discussion by Phaster911** [TanStack/query#4070] [Q&A]
Hello again, I have the following situation:

1. I make a request to get a list of paginated results
2. I have a separate dependent query that needs a collection of ids gathered from the results of the first request
3. I need the data from the second query to be separate from the list data, because I have need of it in separate components that I need the list data, so the list query has a different key than the dependent query

Everything works fine when changing the page by clicking but I wanted to implement prefetching when the user hovers the mouse over a page control, however, you can't have prefetching with dependent queries, at least that's what typescript tells me.
Is there a recipe for this situation or should I just rely on the dependency between queries to make the second request when the list results page does change?

**Comment by TkDodo** [comment]
What is typescript telling you? Which approach have you tried so far? Can you show some code, best in a minimal codesandbox reproduction?
